### ISSUE TITLE
Bug: .ocamlformat has no effect even with --enable-outside-detected-project

### ISSUE THREAD

**Issue by tari3x** [ocaml-ppx/ocamlformat#2223]
```
avatar@pixel:~/docs/visuals/src/spark/lib$ ll ~/.ocamlformat
-rw-r--r-- 1 avatar users 33 Dec 12 00:41 /home/avatar/.ocamlformat
avatar@pixel:~/docs/visuals/src/spark/lib$ ocamlformat --enable-outside-detected-project  server_state.ml
<no effect of settings in ocamlformat>
avatar@pixel:~/docs/visuals/src/spark/lib$ ln -s ~/.ocamlformat 
avatar@pixel:~/docs/visuals/src/spark/lib$ ocamlformat --enable-outside-detected-project  server_state.ml
<works as expected now>
```

I tried renaming ocamlformat to ~/ocamlformat and to ~/.config/ocamlformat and to ~/.config/.ocamlformat. No result. I tried setting XDG_CONFIG_HOME, no result. 

Strace shows that ocamlformat makes no attempt to read any file outside of the current project directory.

**Comment by gpetiot** [comment]
Hi, that's a tough one, I have a few questions.

What version of ocamlformat do you use?

Do you get a warning saying that `--enable-outside-detected-project` is not used? or that no .ocamlformat is found? or that no root is found?

Does your .ocamlformat (or maybe in the parent directory as it seems `lib/` is likely not the project root) contain `disable`? Do you have more .ocamlformat files in your project that could mess up the configuration? (you can check with `ocamlformat --print-config`)

It is recommended to place the .ocamlformat file at the project root unless you want to override the settings in some subdirectories.

**Comment by tari3x** [comment]
```
$ ocamlformat --version 
0.24.1
$ cat .ocamlformat 
profile = janestreet
margin = 75
```

I see no warnings when I run with  --enable-outside-detected-project but I do see a warning when I remove that flag. 

Looks like there are no conflicting config files. 

I have many projects and I would of course want to apply the same formatting in all of them. Thus putting the same .ocamlformat in each of them seems silly. I'd much rather have a single global config.

**Comment by gpetiot** [comment]
Do you have a ocp-indent file in your project?
That would explain the situation: ocamlformat will only look for configuration files outside of the project if no configuration file (.ocamlformat files but also .ocp-indent files) is found in the project. Btw having a symlink pointing to a global .ocamlformat is not a bad solution.

**Comment by tari3x** [comment]
No, I don't have anything with "ocp" in the name. Note that ocamlformat does complain about missing configuration if I omit  --enable-outside-detected-project.

**Comment by tari3x** [comment]
Perhaps we could just add an explicit --config-file argument?

**Comment by jinwoo** [comment]
I also tried this yesterday and found out it didn't work. I'm on M1 macbook and the version of ocamlformat is 0.24.1.

**Comment by Sqvid** [comment]
Just want to confirm that I am experiencing the same bug. No local `ocamlformat` or `ocp*` files.

`version = 0.26.1`
